### PR TITLE
Ensure axis autorange is computed after transform operations

### PR DESCRIPTION
**Pull request by etpinard**
To fix the bug, we need to _clear_ the `ax._min` and `_ax._max` arrays (generated in `Axes.expand` done in the `_module.calc` call before the `calcTransform` loop) before the _final_ `module.calc` call.

To do so in the least _intrusive_ way possible, this PR splits `Plots.doCalcdata` operations into separate loops - which hopefully will also help readability.

### CODE REVIEW

File: src/plots/plots.js [hunk 1]
```diff
@@ -1951,46 +1952,51 @@ plots.doCalcdata = function(gd, traces) {
...
-            continue;
-        }
-
-        var trace = fullData[i],
-            cd = [];
-
-        // If traces were specified and this trace was not included, then transfer it over from
-        // the old calcdata:
         if(Array.isArray(traces) && traces.indexOf(i) === -1) {
             calcdata[i] = oldCalcdata[i];
```
Comment (alexcjohnson) [inline src/plots/plots.js]: I know this part isn't new here, but... it isn't used anywhere, right? Looks to me like it was already broken by transforms (calcdata index and trace index aren't the same) and will be broken MORE by this PR (any trace not included will be dropped from autorange) so I'd remove it until we decide to do it right AND use it.

**Comment by alexcjohnson** [comment]
LGTM. Great tests! nonblocking [comment](https://github.com/plotly/plotly.js/pull/1260#discussion_r93086039), 💃

File: src/plots/plots.js (hunk 1 above)
Comment (etpinard) [inline src/plots/plots.js]: cc @rreusser

Comment (alexcjohnson) [inline src/plots/plots.js]: right, makes a lot of sense to want to do that. I guess for autorange we just need to store the various `_min` and `_max` entries in a different way so we can selectively replace only the ones for the traces we're recalculating. And trace index, depending on how this gets used (it's not yet, right?) you just have to be careful about what you mean by index, before or after transforms potentially insert new traces.

**Comment by VeraZab** [comment]
🎉 
range
this fixes: https://github.com/plotly/streambed/issues/8802
💃 💃 💃

**Comment by etpinard** [comment]
will be part of `1.21.3` set to be released in the week of Jan 2.

File: src/plots/plots.js [hunk 2]
```diff
@@ -1951,46 +1952,51 @@ plots.doCalcdata = function(gd, traces) {
...
+                    _module.calcTransform(gd, trace, transform);
                 }
             }
+        }
+    }
+
+    // clear stuff ...
+    for(i = 0; i < axList.length; i++) {
+        axList[i]._min = [];
+        axList[i]._max = [];
```
Comment (etpinard) [inline src/plots/plots.js]: note to self: maybe we should clear `_categories` too to fix https://github.com/plotly/plotly.js/issues/1181 ?

Comment (alexcjohnson) [inline src/plots/plots.js]: Cool - note though that this only fixes the first part of 1181, not the variant of it mentioned in https://github.com/plotly/plotly.js/issues/1181#issuecomment-262069568

**Comment by alexcjohnson** [comment]
nice. 💃

**Pull request merged**